pkg/serverless/registration/telemetry_api_test.go

```diff
@@ -86,13 +86,18 @@ func TestSubscribeLogsSuccess(t *testing.T) {
 
 func TestSubscribeLogsTimeout(t *testing.T) {
 	payload := buildLogRegistrationPayload("myUri", "platform function", 10, 100, 1000)
+	done := make(chan struct{})
 	//fake the register route
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		// timeout
-		time.Sleep(registerLogsTimeout + 100*time.Millisecond)
+		select {
+		case <-time.After(registerLogsTimeout + 5*time.Second):
+		case <-done:
+		}
 		w.WriteHeader(200)
 	}))
 	defer ts.Close()
+	defer close(done)
```

Comment:
Despite calling `ts.Close()` at the end of the test, it was still waiting on the sleep before finishing. This meant that increasing the timeout also meant increasing the test execution time. I fixed this by allowing the server a faster way out when the test is done.